apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java

```diff
@@ -81,19 +86,33 @@ public String getDeploymentEnvStatus(int id) {
 
     @LoggedIn
     @POST("/deployment")
-    public void addDeployment(int environmentId, int serviceId, int deployableVersionId, String deploymentMessage, Req req) {
-        try {
-            Deployment deployment = deploymentHandler.addDeployment(environmentId, serviceId, deployableVersionId, deploymentMessage, req);
-            assignJsonResponseToReq(req, HttpStatus.CREATED, deployment);
-        } catch (ApolloDeploymentBlockedException e) {
-            assignJsonResponseToReq(req, HttpStatus.FORBIDDEN, e.getMessage());
-        } catch (ApolloDeploymentConflictException e) {
-            assignJsonResponseToReq(req, HttpStatus.CONFLICT, e.getMessage());
-        } catch (ApolloDeploymentTooManyRequestsException e) {
-            assignJsonResponseToReq(req, HttpStatus.NOT_ACCEPTABLE, e.getMessage());
-        } catch (Exception e) {
-            assignJsonResponseToReq(req, HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
-        }
+    public void addDeployment(String environmentIdsCsv, String serviceIdsCsv, int deployableVersionId, String deploymentMessage, Req req) {
+        Iterable<String> environmentIds = Splitter.on(IDS_DELIMITER).omitEmptyStrings().trimResults().split(environmentIdsCsv);
+        Iterable<String> serviceIds = Splitter.on(IDS_DELIMITER).omitEmptyStrings().trimResults().split(serviceIdsCsv);
+
+        MultiDeploymentResponseObject responseObject = new MultiDeploymentResponseObject();
+
+        DeployableVersion deployableVersion = deployableVersionDao.getDeployableVersion(deployableVersionId);
+
+        environmentIds.forEach(environmentIdString -> serviceIds.forEach(serviceIdString -> {
+
+            int environmentId = Integer.parseInt(environmentIdString);
+            int serviceId = Integer.parseInt(serviceIdString);
+
+            DeployableVersion serviceDeployableVersion = deployableVersionDao.getDeployableVersionFromSha(deployableVersion.getGitCommitSha(), serviceId);
+
+            if (serviceDeployableVersion == null) {
+                responseObject.addUnsuccessful(environmentId, serviceId, "There is no deployableVersion for commit sha " + deployableVersion.getGitCommitSha() + " and service " + serviceId);
+            } else {
+                try {
+                    Deployment deployment = deploymentHandler.addDeployment(environmentId, serviceId, serviceDeployableVersion.getId(), deploymentMessage, req);
+                    responseObject.addSuccessful(environmentId, serviceId, deployment);
+                } catch (ApolloDeploymentException e) {
+                    responseObject.addUnsuccessful(environmentId, serviceId, e.getMessage());
+                }
+            }
+        }));
+        assignJsonResponseToReq(req, HttpStatus.CREATED, responseObject);
```

Comment:
This is a bit confusing. Lets just maybe add a verification if all deployments failed, and return something else?
Also, please add a documentation in the wiki regarding this controller and response object